cmd/node/main.go

```diff
@@ -542,6 +550,50 @@ func startNode(ctx *cli.Context, log logger.Logger, version string) error {
 		log.Debug("no epoch db found in storage", "error", errNotCritical.Error())
 	}
 
+	epochFoundInStorage := err == nil
+
+	isCurrentTimeBeforeGenesis := time.Now().Sub(startTime) < 0
```

Comment:
all logic has to be in the epoch start bootstrapper. it will be the default starter mechanism. it gathers data, the rest is continued by the main thread.

Reply:
added those ifs to a 'static' function as it does not make sense to populate the bootstrapper components with many fields it will only used once